sqlmesh/core/model/kind.py

```diff
@@ -81,18 +92,80 @@ def model_kind_name(self) -> ModelKindName:
 class ModelKind(PydanticModel, ModelKindMixin):
     name: ModelKindName
 
-    def to_expression(self, **kwargs: t.Any) -> d.ModelKind:
-        return d.ModelKind(this=self.name.value.upper(), **kwargs)
+    @classmethod
+    def field_validator(cls) -> classmethod:
```

Comment:
why was this change needed?

Reply:
Initially I was adding more field validators for the new FULL_WITH_HISTORY and it was clumsy adding them to the bottom of the file when they might reference a class above it. If that happened then you had to put the definition before that class. This way we couple the validators with the thing they are validating so it is easy to get the appropriate validator for the given field you are referencing. So the need is reduced but I think it is a better pattern.